File: src/main/java/entity/Task.java
package entity;

public class Task {
    private String name;
    private double price;
    private int priority;
    private boolean done;
    private String date;

    public Task(String name, String date, int priority, boolean done) {
        this.name = name;
        this.date = date;
        this.priority = priority;
        this.done = done;
    }

    public Task(String name, String date, int priority, boolean done, double price) {
        this(name, date, priority, done);
        this.price = price;
    }

    public String getName() {
        return name;
    }

    public double getPrice() {
        return price;
    }

    public int getPriority() {
        return priority;
    }

    public boolean isDone() {
        return done;
    }

    public String getDate() {
        return date;
    }

    public void setPrice(double price) {
        this.price = price;
    }
}

File: node/js/task.js
//$(document).ready( () => { })

$(() => {
    console.log("Ready");
    $.ajax({
        url: "http://localhost:3000/task",
        type: "get",

        dataType: "json",

        statusCode: {
            200: (result) => {
                // console.log(result);
                for(var index in result){
                    const id = result[index]._id;
                    const name = result[index].name;
                    const priority = result[index].priority;
                    const date = result[index].date;
                    const done = result[index].done;
                    const price = result[index].price;
                    var text = done?"<label class=\"task_text_done\"> ":"<label class=\"task_text\"> ";
                    text = text + "<b>" + name + "</b> <u>(" + date + ")</u> ";
                    text = text + "Priority: " + priority + "<br>";
                    if (result[index].price) text = text + "Price: " + price + "<br>";
                    text = text + "Done: " + done + "<br>";
                    if (!done) text = text + "<button id=" + id + " onClick=\"complete(this.id)\"> Complete";
                    text = text + "</label>";
                    $("#parent").append(text);
                }
            },
            400: (err) => {
                console.log("Bad request");
            },
            404: (err) => {
                console.log("Not found");
            }
        },
        
    })
});

function complete(id){
    id = {"_id": id};
    $.ajax({
        url: "http://localhost:3000/task/done",
        type: "put",
        data: id,
        statusCode: {
            200: location.reload()
        }
    });
}